### PR TITLE
feat(swagger-ui-react): rewrite into SSR compatible function component

### PR DESCRIPTION
**Pull request by char0n**
Refs #9243

### Description

This is a rewrite of `swagger-ui-react` from class component into function component. Class components are starting to be completely unsupported by the modern browsers.

### Motivation and Context

#9243

### How Has This Been Tested?

Using https://github.com/char0n/swagger-ui-nextjs

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

---

**Resources:**

- [use client](https://react.dev/reference/react/use-client#use-client) explanation

### CODE REVIEW

File: flavors/swagger-ui-react/index.jsx [hunk 1]
```diff
@@ -1,101 +1,148 @@
...
+    .displayRequestDuration,
+  withCredentials = SwaggerUIConstructor.defaultOptions.withCredentials,
+  persistAuthorization = SwaggerUIConstructor.defaultOptions
+    .persistAuthorization,
+  oauth2RedirectUrl = SwaggerUIConstructor.defaultOptions.oauth2RedirectUrl,
+  onComplete = null,
+}) => {
+  const systemRef = useRef(null)
+  const SwaggerUIComponentRef = useRef(null)
+  const [, forceUpdate] = useState(undefined)
```
Comment (char0n) [inline flavors/swagger-ui-react/index.jsx:48]:
```suggestion
  const [, forceUpdate] = useState(false)
```

File: flavors/swagger-ui-react/index.jsx [hunk 2]
```diff
@@ -1,101 +1,148 @@
...
 import SwaggerUIConstructor from "#swagger-ui"
 
-class SwaggerUI extends React.Component {
-  constructor (props) {
-    super(props)
-    this.SwaggerUIComponent = null
-    this.system = null
-  }
+const SwaggerUI = ({
+  spec = SwaggerUIConstructor.defaultOptions.url,
```
Comment (char0n) [inline flavors/swagger-ui-react/index.jsx:10]:
```suggestion
  spec = SwaggerUIConstructor.defaultOptions.spec,
```

**Review by glowcloud — APPROVED** [review]
Code looks good to me, not seeing anything missing. 

I also ran projects with React and Next.js and had no errors. There are only warnings that I saw there before: `Failed to parse source map` for some modules in React and `UNSAFE_componentWillReceiveProps` in both, but the second one is just for the future rewrite of class components.

**Comment by char0n** [comment]
Verified that babel tranpisaltion doesn't remove `"use client"` directive.

File: flavors/swagger-ui-react/index.jsx [hunk 3]
```diff
@@ -1,101 +1,137 @@
-import React from "react"
+/**
+ * @prettier
+ */
+"use client"
```
Comment (char0n) [inline flavors/swagger-ui-react/index.jsx:4]:
```suggestion
"use client"

```

**Pull request merged**